packages/cli/src/oclif/commands/scaffold.js

```diff
@@ -0,0 +1,217 @@
+    }
+
+    const templateFile = path.join(
+      __dirname,
+      '../../..',
+      `scaffold/${type}.template.js`
+    );
+    const testTemplateFile = path.join(
+      __dirname,
+      '../../..',
+      'scaffold/test.template.js'
+    );
+
+    const { dest = destMap[type], entry = 'index.js' } = this.flags;
+    const entryFile = path.join(process.cwd(), entry);
+
+    this.log(`Adding ${type} scaffold to your project.\n`);
+
+    return this.writeTemplateFile(templateFile, templateContext, dest)
+      .then(() =>
+        this.writeTemplateFile(
+          testTemplateFile,
+          templateContext,
+          `test/${dest}`
+        )
+      )
+      .then(() => utils.readFile(entryFile))
+      .then(entryBuf => entryBuf.toString())
+      .then(entryJs => {
+        this.startSpinner(`Rewriting your ${entry}`);
+        const lines = entryJs.split('\n');
+
+        // this is very dumb and will definitely break, it inserts lines of code
+        // we should look at jscodeshift or friends to do this instead
+
+        // insert Resource = require() line at top
+        const varName = `${templateContext.CAMEL}${utils.camelCase(type)}`;
+        const importerLine = `const ${varName} = require('./${dest}');`;
+        lines.splice(0, 0, importerLine);
+
+        // insert '[Resource.key]: Resource,' after 'resources:' line
+        const injectAfter = `${typeMap[type]}: {`;
+        const injectorLine = `[${varName}.key]: ${varName},`;
+        const linesDefIndex = lines.findIndex(line =>
+          line.endsWith(injectAfter)
+        );
+
+        if (linesDefIndex === -1) {
+          this.stopSpinner(false);
+          this.log();
+          this.log(
+            colors.bold(`Oops, we could not reliably rewrite your ${entry}.`) +
+              ' Please add:'
+          );
+          this.log(` * \`${importerLine}\` to the top`);
+          this.log(
+            ` * \`${injectAfter} ${injectorLine} },\` in your app definition`
+          );
+          return Promise.resolve();
+        } else {
+          lines.splice(linesDefIndex + 1, 0, '    ' + injectorLine);
+          return utils
+            .writeFile(entryFile, lines.join('\n'))
+            .then(this.stopSpinner);
+        }
+      })
+      .then(() =>
+        this.log(
+          '\nFinished! We did the best we could, you might gut check your files though.'
+        )
+      )
+      .catch(message => {
+        this.stopSpinner(false);
+        this.log(colors.red(`We couldn't scaffold your files:`));
+        this.log(message);
+      });
+  }
+}
+
+ScaffoldCommand.args = [
+  {
+    name: 'type',
+    help: 'what type of thing are you creating',
+    required: true,
+    options: ['resource', 'trigger', 'search', 'create']
+  },
+  {
+    name: 'name',
+    help: 'the name of the new thing to create',
+    required: true
+  }
+];
+
+ScaffoldCommand.flags = buildFlags({
+  commandFlags: {
+    dest: flags.string({
+      description:
```

Comment:
i'd probably add `char`s to these other flags. `-d` and `-e`, respectively. We could even set `buildFlags` to do that automatically, if we wanted to be fancy.